tests/unit/validator.js

```diff
@@ -80,47 +88,87 @@ tap.test('Validation Middleware', {autoend: true}, t => {
       t.end()
     })
 
+    t.test('should validate lookupRequests data', t => {
+      const ctx = {
+        request: {},
+        lookupRequests: {
+          name: 'Typer',
+          surname: 'Durden'
+        },
+        state: {
+          metaData: {
+            inputValidation: {
+              type: 'object',
+              properties: {
+                lookupRequests: {
+                  type: 'object',
+                  properties: {
+                    name: {type: 'string'},
+                    surname: {type: 'string'}
+                  },
+                  required: ['name']
+                }
+              }
+            }
+          }
+        }
+      }
+
+      t.doesNotThrow(() => performValidation(ctx))
+      t.end()
+    })
+
     t.test('should validate', t => {
+      t.plan(2)
+
       const ctx = {
         request: {
           body: {
-            name: 'drake',
-            surname: 'durden'
+            name: 'Drake',
+            surname: 'Durden'
           }
         },
         state: {
           metaData: {
-            type: 'object',
-            properties: {
-              requestBody: {
-                type: 'object',
-                properties: {
-                  name: {type: 'string'},
-                  surname: {type: 'string'}
-                },
-                required: ['name']
+            inputValidation: {
+              type: 'object',
+              properties: {
+                requestBody: {
+                  type: 'object',
+                  properties: {
+                    name: {type: 'string'},
+                    surname: {type: 'string'}
+                  },
+                  required: ['name']
+                }
               }
             }
           }
         }
       }
 
-      t.doesNotThrow(() => performValidation(ctx))
+      t.doesNotThrow(() =>
+        validateBodyMiddleware()(ctx, () => {
+          t.pass()
+        })
+      )
       t.end()
     })
 
     t.test('should validate when a property has a value of null', t => {
-      // clear the module with the out-dated config
-      delete require.cache[require.resolve('../../src/middleware/validator')]
-      // set environment variable to allow null values
+      // // clear the module with the out-dated config
+      delete require.cache[require.resolve('rewire')]
+      // // set environment variable to allow null values
       process.env.VALIDATION_ACCEPT_NULL_VALUES = 'true'
-      // require module to get updated config
-      const validatorUpdatedEnv = require('../../src/middleware/validator')
+      // // require module to get updated config
+      const rewire = require('rewire')
+      const validator = rewire('../../src/middleware/validator')
+      const performValidation = validator.__get__('performValidation')
```

Comment:
Can we move these require files to the top of the script instead if directly in the test please, unless there is a specific reason for this?

Reply:
I delete the `rewire` require cache above to reset the config. It's needed there

Reply:
aah right i see, you adding a different setting to AJV so it needs to be required again for it to be applied :+1: